android/src/main/java/com/actiotech/twiliovideorn/TwilioVideoModule.java

```diff
@@ -1066,4 +1067,37 @@ public void onMessage(RemoteDataTrack remoteDataTrack, String message) {
             }
         };
     }
+
+    @Override
+    public void onStats(List<StatsReport> statsReports) {
+        WritableMap event = new WritableNativeMap();
+        for (StatsReport sr : statsReports) {
+            WritableMap connectionStats = new WritableNativeMap();
+            WritableArray as = new WritableNativeArray();
+            for (RemoteAudioTrackStats s : sr.getRemoteAudioTrackStats()) {
+                as.pushMap(convertAudioTrackStats(s));
+            }
+            connectionStats.putArray("remoteAudioTrackStats", as);
+
+            WritableArray vs = new WritableNativeArray();
+            for (RemoteVideoTrackStats s : sr.getRemoteVideoTrackStats()) {
+                vs.pushMap(convertVideoTrackStats(s));
+            }
+            connectionStats.putArray("remoteVideoTrackStats", vs);
+
+            WritableArray las = new WritableNativeArray();
+            for (LocalAudioTrackStats s : sr.getLocalAudioTrackStats()) {
+                las.pushMap(convertLocalAudioTrackStats(s));
+            }
+            connectionStats.putArray("localAudioTrackStats", las);
+
+            WritableArray lvs = new WritableNativeArray();
+            for (LocalVideoTrackStats s : sr.getLocalVideoTrackStats()) {
+                lvs.pushMap(convertLocalVideoTrackStats(s));
+            }
+            connectionStats.putArray("localVideoTrackStats", lvs);
+            event.putMap(sr.getPeerConnectionId(), connectionStats);
+        }
```

Comment:
It might be a premature optimization, but do we use all these stats all the time?
Building this list many times per second would likely drain the battery. Would it be helpful to subscribe to specific kinds of stats (Eg. audioTrack only)?